packages/api-plugin-fulfillment/src/simpleSchemas.js

```diff
@@ -0,0 +1,169 @@
+import SimpleSchema from "simpl-schema";
+
+/**
+ * @name MethodEmptyData
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Empty placeholder
+ * @property {String} gqlType Defines the method type
+ * @property {Boolean} emptyData Empty Data fields
+ */
+export const MethodEmptyData = new SimpleSchema({
+  gqlType: String,
+  emptyData: {
+    type: Boolean,
+    optional: true
+  }
+});
+
+/**
+ * @name FulfillmentMethodSchema
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Schema for Fulfillment Method
+ * @property {Number} cost - optional cost
+ * @property {String[]} fulfillmentTypes - fulfillment type retained for backward compatibility
+ * @property {String} group - Group to which fulfillment method belong
+ * @property {Number} handling - handling charges
+ * @property {Boolean} enabled - status of fulfillment method
+ * @property {String} label - displayed on the UI
+ * @property {String} name - name of the fulfillment method, user editable
+ * @property {String} fulfillmentMethod - used by application, not user editable
+ * @property {String} displayMessageMethod - used to display additional message on UI specific to ff-method
+ * @property {Number} rate - ratefor themethod
+ */
+export const FulfillmentMethodSchema = new SimpleSchema({
+  "cost": {
+    type: Number,
+    optional: true
+  },
+  "fulfillmentTypes": {
+    type: Array,
+    minCount: 1
+  },
+  "fulfillmentTypes.$": String,
+  "group": String,
+  "handling": Number,
+  "enabled": Boolean,
+  "label": String,
+  "name": String,
+  "fulfillmentMethod": {
+    type: String,
+    optional: true
+  },
+  "displayMessageMethod": {
+    type: String,
+    optional: true
+  },
+  "rate": Number
+});
+
+/**
+ * @name ProviderSchema
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Schema for Provider details inside Fulfillment Type
+ * @property {Boolean} enabled - status of fulfillment method
+ * @property {String} label - displayed on the UI
+ * @property {String} name - name of the fulfillment type, user editable
+ */
+const ProviderSchema = new SimpleSchema({
+  enabled: Boolean,
+  label: String,
+  name: String
+});
+
+/**
+ * @name FulfillmentTypeSchema
+ * @memberof Schemas
+ * @type {SimpleSchema}
+ * @summary Defines Schema for Fulfillment Type
+ * @property {String} name - name of the fulfillment type, user editable
+ * @property {String} shopId - Shop ID
+ * @property {ProviderSchema} provider - Provider details
+ * @property {String} fulfillmentType - fulfillmentType, not user editable
+ * @property {String} displayMessageType - used to display additional message on UI specific to ff-type
+ * @property {FulfillmentMethodSchema[]} methods - array of ff-methods under this ff-type
+ */
+export const FulfillmentTypeSchema = new SimpleSchema({
```

Comment:
All new collections are supposed to have `createdAt` and `updatedAt` properties. You'd need to add in the SimpleSchema, in GraphQL, and also set them properly in the create/update mutations. If you look at newer code in other plugins you should be able to find examples of this.

Reply:
@aldeed @zenweasel 
I have made the changes to SimpleSchema to add the createdAt & updatedAt for the ff-type at root level of the doc.
I have not added in GraphQL since the createFFType mutation is not public and the updateFFType we populate the updatedAt in the code instead of user-input

Question:
We have `methods` array under each ff-type. Do we need to introduce the createdAt & updatedAt for each ff-method as well?

Reply:
> We have methods array under each ff-type. Do we need to introduce the createdAt & updatedAt for each ff-method as well?

No. These fields are at the collection level only

Reply:
@zenweasel Thanks for clarifying.